src/components/App.tsx

```diff
@@ -1578,10 +1579,12 @@ class App extends React.Component<AppProps, AppState> {
     }
     this.setState((prevState) => {
       return {
-        elementLocked: !prevState.elementLocked,
-        activeTool: prevState.elementLocked
-          ? { type: "selection" }
-          : prevState.activeTool,
+        activeTool: {
+          locked: !prevState.activeTool.locked,
+          type: prevState.activeTool.locked
+            ? "selection"
+            : prevState.activeTool.type,
+        },
```

Comment:
this overwrites `lastToolBeforeEraser` or what it's called. I'm thinking we should make that prop required (but nullable)

Reply:
done 👍🏻